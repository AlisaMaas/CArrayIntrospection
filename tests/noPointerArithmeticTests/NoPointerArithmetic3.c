#include <stdio.h>
/**
* This tests that a simple loop with an updated pointer is corrected as expected.
**/
int find2(int* x, int y, int me){
  int *end = x + y;
  for(; x != end; x++){
    if(*x == me)
      return me;
  }
  return -1;
}