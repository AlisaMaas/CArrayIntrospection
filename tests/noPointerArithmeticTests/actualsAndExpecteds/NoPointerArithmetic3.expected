Printing analysis 'Promote Memory to Register' for function 'find2':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Replace pointer arithmetic with GEP instructions':

entry:
  %idx.ext = sext i32 %y to i64
  %add.ptr = getelementptr inbounds i32* %x, i64 %idx.ext
  br label %for.cond

for.cond:                                         ; preds = %for.inc, %entry
  %0 = phi i32 [ 0, %entry ], [ %3, %for.inc ]
  %1 = getelementptr i32* %x, i32 %0
  %cmp = icmp ne i32* %1, %add.ptr
  br i1 %cmp, label %for.body, label %for.end

for.body:                                         ; preds = %for.cond
  %2 = load i32* %1, align 4
  %cmp1 = icmp eq i32 %2, %me
  br i1 %cmp1, label %if.then, label %if.end

if.then:                                          ; preds = %for.body
  br label %return

if.end:                                           ; preds = %for.body
  br label %for.inc

for.inc:                                          ; preds = %if.end
  %3 = add nsw i32 %0, 1
  %incdec.ptr = getelementptr inbounds i32* %1, i32 1
  br label %for.cond

for.end:                                          ; preds = %for.cond
  br label %return

return:                                           ; preds = %for.end, %if.then
  %retval.0 = phi i32 [ %me, %if.then ], [ -1, %for.end ]
  ret i32 %retval.0
